src/xdplwf/recv.c

```diff
@@ -496,45 +531,46 @@ XdpGenericReceiveEnqueueTxNb(
     // TODO: Perform any software offloads required.
     //
 
-    if (CanPend) {
+    if (RxQueue->TxCloneNblList == NULL) {
+        RxQueue->TxCloneNblList =
+            (NET_BUFFER_LIST *)InterlockedFlushSList(&RxQueue->TxCloneNblSList);
+    }
+
+    if (RxQueue->TxCloneNblList != NULL) {
+        TxNbl = RxQueue->TxCloneNblList;
```

Comment:
Do we need to nullify the Next field on TxNbl?

Reply:
We shouldn't need to, because `NdisAppendSingleNblToNblCountedQueue` below automatically NULLs the `Next` field, and the error cases also overwrite the `Next` field. It might be a good idea to NULL it, but many of our NBL routines are explicitly single-NBL and therefore `Next` is required to be ignored.